spec/ransack/search_spec.rb

```diff
@@ -234,12 +234,21 @@ module Ransack
           parent_name_eq: "parent_name_query",
           parent_articles_title_eq: 'parents_article_title_query'
         }).result
+
         real_query = remove_quotes_and_backticks(s.to_sql)
 
-        expect(real_query)
-          .to match(%r{LEFT OUTER JOIN articles ON (\('default_scope' = 'default_scope'\) AND )?articles.person_id = people.id})
-        expect(real_query)
-          .to match(%r{LEFT OUTER JOIN articles articles_people ON (\('default_scope' = 'default_scope'\) AND )?articles_people.person_id = parents_people.id})
+        if ::Gem::Version.new(::ActiveRecord::VERSION::STRING) > ::Gem::Version.new(Constants::RAILS_6_0)
```

Comment:
Can we also add an a test where the data for this condition is created, and we check that the output is the same across each version of AR.

Reply:
I have tried but i couldn't find a way to change the dynamically change the version of AR. But i have tested with different versions without spec, it's fine.

Reply:
Thanks @shadabmalik1310 the way to do this would to:

- set up some data 
- check for an expected output of a Ransack query as data (not the SQL generated)
- run it in CI for each different version of Rails